src/contest.c

```diff
@@ -2740,3 +2791,3403 @@ bool8 sub_80DB798(u8 a)
+    strb r0, [r1, 0x5]\n\
+    ldr r0, [r4]\n\
+    ldr r0, [r0, 0x14]\n\
+    adds r3, r0\n\
+    ldrb r0, [r3, 0x1]\n\
+    lsls r1, r0, 4\n\
+    adds r1, r0\n\
+    lsls r1, 2\n\
+    adds r1, r7\n\
+    ldrb r2, [r1, 0x5]\n\
+    mov r0, r8\n\
+    ands r0, r2\n\
+    orrs r0, r6\n\
+    strb r0, [r1, 0x5]\n\
+    adds r5, 0x1\n\
+    cmp r5, 0x3\n\
+    ble _080DE2F0\n\
+    add sp, 0xC\n\
+    pop {r3}\n\
+    mov r8, r3\n\
+    pop {r4-r7}\n\
+    pop {r0}\n\
+    bx r0\n\
+    .pool\n\
+    .syntax divided\n");
+}
+#endif
+
+#ifdef NONMATCHING
+void sub_80DE350(void)
+{
+    s32 i;
+    s16 var;
+    u16 bg1Cnt;
+    u8 *spriteID;
+    u16 something;
+
+    RequestDma3Fill(0,(void *)(VRAM + 0x8000), 0x2000, 0x1);
+    //DmaClearLarge32(3, (void *)(VRAM + 0x8000), 0x2000, 0x1000);
+    
+    /*bg1Cnt = 0;
+    something = 0;
+    var = bg1Cnt;
+    spriteID = (u8 *)&gContestResources;*/
+    
+    var = 0;
+    
+    
+    CpuSet(&var, &gContestResources->field_24[1], 0x05000400);
+    //CpuSet(src, (void *)(VRAM + 0x10000 + (gSprites[sContest.applauseMeterSpriteId].oam.tileNum + 17 + i)
+    Contest_SetBgCopyFlags(1);
+    
+    //DmaClear32(3, (void *)(VRAM + 0xF000), 0x1000);
+    
+    bg1Cnt = GetGpuReg(REG_OFFSET_BG1CNT);
+    something = bg1Cnt;
+    ((struct BgCnt *) &bg1Cnt)->priority = 1;
+    ((struct BgCnt *) &bg1Cnt)->screenSize = 1;
+    ((struct BgCnt *) &bg1Cnt)->areaOverflowMode = 0;
+    ((struct BgCnt *) &bg1Cnt)->charBaseBlock = 2;
+    
+    SetGpuReg(REG_OFFSET_BG1CNT, bg1Cnt);
+
+    gBattle_BG1_X = 0;
+    gBattle_BG1_Y = 0;
+
+    for (i = 0; i < 4; i++)
+    {
+        gSprites[gContestResources->field_14[i].unk0].oam.priority = 0;
+        gSprites[gContestResources->field_14[i].unk1].oam.priority = 0;
+    }
+}
+#else
+NAKED
+void sub_80DE350(void)
+{
+    asm(".syntax unified\n\
+    push {r4-r7,lr}\n\
+    sub sp, 0x8\n\
+    ldr r1, =0x06008000\n\
+    movs r2, 0x80\n\
+    lsls r2, 6\n\
+    movs r0, 0\n\
+    movs r3, 0x1\n\
+    bl RequestDma3Fill\n\
+    movs r5, 0\n\
+    str r5, [sp]\n\
+    ldr r6, =gContestResources\n\
+    ldr r0, [r6]\n\
+    ldr r1, [r0, 0x28]\n\
+    ldr r2, =0x05000400\n\
+    mov r0, sp\n\
+    bl CpuSet\n\
+    movs r0, 0x1\n\
+    bl Contest_SetBgCopyFlags\n\
+    movs r0, 0xA\n\
+    bl GetGpuReg\n\
+    add r3, sp, 0x4\n\
+    strh r0, [r3]\n\
+    ldrb r1, [r3]\n\
+    movs r0, 0x4\n\
+    negs r0, r0\n\
+    ands r0, r1\n\
+    movs r1, 0x1\n\
+    orrs r0, r1\n\
+    strb r0, [r3]\n\
+    mov r1, sp\n\
+    adds r1, 0x5\n\
+    ldrb r2, [r1]\n\
+    movs r0, 0x3F\n\
+    ands r0, r2\n\
+    strb r0, [r1]\n\
+    ldrb r2, [r1]\n\
+    movs r0, 0x21\n\
+    negs r0, r0\n\
+    ands r0, r2\n\
+    strb r0, [r1]\n\
+    ldrb r1, [r3]\n\
+    movs r4, 0xD\n\
+    negs r4, r4\n\
+    adds r0, r4, 0\n\
+    ands r0, r1\n\
+    movs r1, 0x8\n\
+    orrs r0, r1\n\
+    strb r0, [r3]\n\
+    ldrh r1, [r3]\n\
+    movs r0, 0xA\n\
+    bl SetGpuReg\n\
+    ldr r0, =gBattle_
```

Comment:
This will most likely match with an upwards loop, the compiler just optimized it to use a downwards one